homework/index.js

```diff
@@ -27,21 +27,159 @@
+          });
+          createAndAppend('span', avatarContainer2, {
+            text: numberOfContributor,
+            class: 'result label',
+          });
+          createAndAppend('br', avatarContainer2, {});
+        }
+      }
+    });
+  }
+
+  function reloadSelectedRepository(valueOfSelectedRepository, selectedData) {
+    const resetInfoDiv = document.getElementById('info');
+    const resetContDiv = document.getElementById('cont');
+    resetInfoDiv.innerHTML = '';
+    resetContDiv.innerHTML = '';
+    const getDescription = selectedData[valueOfSelectedRepository].description;
+    const getForks = selectedData[valueOfSelectedRepository].forks;
+    const getUpdatedAt = selectedData[valueOfSelectedRepository].updated_at;
+    const dataName = selectedData[valueOfSelectedRepository].name;
+
+    // next line is an another url and need fetch to get contributors
+    const getContributorsUrl = selectedData[valueOfSelectedRepository].contributors_url;
+
+    if (getDescription == null) {
+      const noDescription = createAndAppend('li', resetInfoDiv, {});
+      createAndAppend('span', noDescription, {
+        text: 'Description: there is no description',
+        class: '',
+      });
+    } else {
+      const liDescription = createAndAppend('li', resetInfoDiv, {});
+      createAndAppend('span', liDescription, { text: 'Description: ' });
+      createAndAppend('span', liDescription, { text: getDescription, class: 'result' });
+    }
+
+    const link = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', link, { text: 'link: ' });
+    createAndAppend('a', link, {
+      text: dataName,
+      target: '_blank',
+      href: `https://github.com/HackYourFuture/${dataName}`,
+      class: 'result',
+    });
+    const liOfForks = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', liOfForks, { text: 'Forks: ' });
+    createAndAppend('span', liOfForks, { text: getForks, class: 'result' });
+    const liUpdateAt = createAndAppend('li', resetInfoDiv, {});
+    createAndAppend('span', liUpdateAt, { text: 'updated_at: ' });
+    createAndAppend('span', liUpdateAt, { text: getUpdatedAt, class: 'result' });
+
+    // call all contributors for this selected repository
+    fetchContributorsUrl(getContributorsUrl); // comment this line to stop getting contributors
+  }
+
+  function reloadAllRepositories(parentElement, data) {
+    const pre = document.getElementById('pre');
+    const parent = createAndAppend('div', parentElement, { class: 'bodyInformation' });
+    createAndAppend('div', parent, { id: 'info', class: 'info' });
+    createAndAppend('div', parent, { id: 'cont', class: 'cont' });
+
+    const selectRepository = createAndAppend('select', pre, { id: 'select', class: 'select' });
+
+    for (let i = 0; i < data.length; i++) {
+      createAndAppend('option', selectRepository, { text: data[i].name, value: i });
+    }
+    // sort list of repository's
+    sortList();
```

Comment:
As mentioned above, you could sort the data before rendering, it is simpler.

    const reposSorted = data.sort((r1, r2) => r1.name.localeCompare(r2.name));
    for (let i = 0; i < reposSorted.length; i++) {
      createAndAppend('option', selectRepository, { text: reposSorted[i].name, value: i });
    }